support/util/pausereconcile.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+import (
+	"strconv"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+)
+
+// IsReconciliationPaused checks the pause reconciliation annotation to see if reconciliation on the resource should be
+// paused. It can either be a date in RFC3339 format or a boolean. If it is a date: reconciliation will be paused until
+// that date. If it is a true boolean value: it will be paused until the annotation is removed.
+func IsReconciliationPaused(annotations map[string]string) bool {
+	val, ok := annotations[hyperv1.PauseReconciliationAnnotation]
+	if !ok {
+		return false
```

Comment:
this func should either return error or log it, otherwise if a typo in the annotation this returns false with no clue why

Reply:
adding logging support

Reply:
added